### PR TITLE
Fix UiProvider - unique investments

### PR DESCRIPTION
**Pull request by ruslanpalagin**

### CODE REVIEW

File: contracts_arbitrum/contracts/v3/RewardsDistributor.sol [hunk 1]
```diff
@@ -188,6 +188,11 @@ contract RewardsDistributor is IRewardsDistributor {
     require(pool.lastRewardTime > 0, "Pool not found");
     _updatePool(msg.sender);
     UserInfo storage user = userInfo[msg.sender][_user];
+
+    if (user.amount == 0) {
```
Comment (boivlad) [inline contracts_arbitrum/contracts/v3/RewardsDistributor.sol:192]: What happens if the user sells all the tokens, then buys again, it also debounces his address

File: frontend/app/shared/coreManager/smartContractsFactory.ts [hunk 2]
```diff
@@ -14,6 +14,8 @@ export const $coreManagerSmartContractSigned = atom<RewardsDistributor | null>((
   if (!rpcConfig || !provider) {
     return null
   }
+  console.log('rpcConfig', rpcConfig)
   const smartContract = new ethers.Contract(rpcConfig.contractsAddresses.coreManagerAddress, coreManagerAbi, provider);
+  console.log('provider', provider.getSigner().getAddress())
```
Comment (boivlad) [inline frontend/app/shared/coreManager/smartContractsFactory.ts:19]: I think this console can remove

File: frontend/app/shared/coreManager/smartContractsFactory.ts [hunk 3]
```diff
@@ -14,6 +14,8 @@ export const $coreManagerSmartContractSigned = atom<RewardsDistributor | null>((
   if (!rpcConfig || !provider) {
     return null
   }
+  console.log('rpcConfig', rpcConfig)
```
Comment (boivlad) [inline frontend/app/shared/coreManager/smartContractsFactory.ts:17]: I think this console can remove

File: frontend/deploys/localhost.json [hunk 4]
```diff
@@ -1,14 +1,10 @@
...
-  "RewardsDistributor": "0x681e48f3470b12e65abE22f4d116aC3cc5396C4c",
-  "UiProvider": "0xbE444fE5776a16cD1527aa3bFba3d57f4908a81D"
+  "Usdt": "0xefAB0Beb0A557E452b398035eA964948c750b2Fd",
+  "SeriesMaster": "0x045857BDEAE7C1c7252d611eB24eB55564198b4C",
+  "KycStore": "0x071586BA1b380B00B793Cc336fe01106B0BFbE6D",
+  "AddressesProvider": "0xaca81583840B1bf2dDF6CDe824ada250C1936B4D",
+  "CoreManager": "0xA56F946D6398Dd7d9D4D9B337Cf9E0F68982ca5B",
+  "TokenPriceOracle": "0x71089Ba41e478702e1904692385Be3972B2cBf9e",
+  "RewardsDistributor": "0x02df3a3F960393F5B349E40A599FEda91a7cc1A7",
+  "UiProvider": "0xe70f935c32dA4dB13e7876795f1e175465e6458e"
```
Comment (boivlad) [inline frontend/deploys/localhost.json:9]: Thx you for cleaning)

**Review by boivlad — APPROVED** [review]

File: contracts_arbitrum/contracts/v3/RewardsDistributor.sol (hunk 1 above)
Comment (ruslanpalagin) [reply contracts_arbitrum/contracts/v3/RewardsDistributor.sol]: +

**Review by dmitriy-green — APPROVED** [review]

**Pull request merged**